cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -821,6 +810,49 @@ private void configureCustomAuthorization(KafkaAuthorizationCustom authorization
         }
     }
 
+    /**
+     * Configures the Kafka configuration providers
+     *
+     * @param userConfig    The user configuration to extract the possible user-provided config provider configuration
+     *                      from it
+     */
+    private void configProviders(KafkaConfiguration userConfig)    {
+        printSectionHeader("Config providers");
+
+        String strimziConfigProviders;
+        if (node.broker()) {
+            // File and Directory providers are used only on broker nodes
+            strimziConfigProviders = "strimzienv,strimzifile,strimzidir";
+        } else {
+            strimziConfigProviders = "strimzienv";
+        }
+
+        if (userConfig != null
+                && !userConfig.getConfiguration().isEmpty()
+                && userConfig.getConfigOption("config.providers") != null) {
+            writer.println("# Configuration providers configured by the user and by Strimzi");
+            writer.println("config.providers=" + userConfig.getConfigOption("config.providers") + "," + strimziConfigProviders);
+            userConfig.removeConfigOption("config.providers");
+        } else {
+            //System.out.println("User config is: " + userConfig.getConfiguration());
+            //System.out.println("User config is: " + userConfig.getConfigOption("config.providers"));
+            writer.println("config.providers=" + strimziConfigProviders);
```

Comment:
Maybe adding a comment on the top like in the other "if" branch, something like:

```java
writer.println("# Configuration providers configured by Strimzi");
```